docs/grammar/SolidityLexer.g4

```diff
@@ -33,7 +33,7 @@ Event: 'event';
 External: 'external';
 Fallback: 'fallback';
 False: 'false';
-Fixed: 'fixed' | ('fixed' [0-9]+ 'x' [0-9]+);
+Fixed: 'fixed' | ('fixed' [1-9][0-9]* 'x' [1-9][0-9]*);
```

Comment:
I did the same, but it is more complex than that. We allow 0 after `x`. However we cannot allow consecutive leading zeroes.

Also there's `ufixed` further down the file.

Reply:
It seems to have worked.

Reply:
After all, this is a non-finalized feature...

Reply:
Can you clarify on if `0` is indeed allowed as `N` the `x` for `fixedMxN`? The comments in the updated code explicitly mention that `N` is allowed to be zero, yet this isn't represented in this grammar. What are the semantics of `N = 0`?

Reply:
Since fixed point types are not yet implemented, it is mostly irrelevant what is allowed and what is not, or are you interested in which nonn-keyword identifiers are supported?

Reply:
I think it's important that the grammar reflects what the code supports, do you agree?

Reply:
The grammar currently reflects what the parser supports. Would you prefer that `fixed` is removed from the grammar and the parser at the same time?

Reply:
This file seems to be `docs/grammar/SolidityLexer.g4`, which suggests this is the grammar for what is an accepted Lexeme/Token in the Solidity language, along with how it is classified. Therefore, it should match the behaviour of Solidity's Scanner.cpp and related files. [This line](https://github.com/ethereum/solidity/pull/10580/files#diff-ff7b1158c474fc549268aa31d4d20125f418557bd146dad35e090b16f57f5d4bR207) supports `0` for `N` values in `fixedMxN`, thus I believe this justifies a change to SolidityLexer.g4.

The online soliditylang.org javascript compiler does not warn or error on using `fixed8x0 test;` in a contract body, thus it appears that the parser actually accepts this code (though perhaps that parser is out-of-date). If that is out of date, then can you clarify if the parser rejects fixed point integer types, or if the type checker rejects fixed point integer types?